build/build.js

```diff
@@ -1,28 +1,101 @@
 #!/usr/bin/env node
```

Comment:
the build script has seen a lot of evolutions:

*    proper option parsing with minimist
*    a lot of information is fetched directly from the manifest
*    the update file has been converted to the json format (see https://developer.mozilla.org/en-US/Add-ons/Updates) and most of its information come from the manifest



Still missing:

*    signing; but because we didn't have it before I considered it was not needed in this PR